pgmpy/factors/CPD.py

```diff
@@ -125,6 +125,79 @@ def __repr__(self):
+│ grade_3 │ 0.4     │ 0.9     │ 0.5     │ 0.7     │ 0.3     │ 0.2     │ 0.8     │ 0.8     │ 0.1     │ 0.6     │ 0.8     │ 0.8     │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ grade_4 │ 0.5     │ 0.1     │ 0.6     │ 0.6     │ 0.2     │ 0.1     │ 0.9     │ 0.9     │ 0.8     │ 0.5     │ 0.8     │ 0.8     │
+╘═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╛
+        >>> final_cpd.set_parents_order(['iq', 'intel', 'diff'])
+        >>> print(final_cpd)
+╒═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╕
+│ diff    │ diff_0  │ diff_0  │ diff_0  │ diff_0  │ diff_1  │ diff_1  │ diff_1  │ diff_1  │ diff_2  │ diff_2  │ diff_2  │ diff_2  │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ intel   │ intel_0 │ intel_0 │ intel_1 │ intel_1 │ intel_0 │ intel_0 │ intel_1 │ intel_1 │ intel_0 │ intel_0 │ intel_1 │ intel_1 │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ iq      │ iq_0    │ iq_1    │ iq_0    │ iq_1    │ iq_0    │ iq_1    │ iq_0    │ iq_1    │ iq_0    │ iq_1    │ iq_0    │ iq_1    │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ grade_0 │ 0.12    │ 0.5     │ 0.6     │ 0.5     │ 0.2     │ 0.4     │ 0.9     │ 0.9     │ 0.5     │ 0.8     │ 0.4     │ 0.8     │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ grade_1 │ 0.12    │ 0.5     │ 0.6     │ 0.5     │ 0.2     │ 0.4     │ 0.9     │ 0.9     │ 0.5     │ 0.8     │ 0.4     │ 0.8     │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ grade_2 │ 0.3     │ 0.7     │ 0.8     │ 0.7     │ 0.4     │ 0.2     │ 0.8     │ 0.8     │ 0.8     │ 0.8     │ 0.8     │ 0.4     │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ grade_3 │ 0.4     │ 0.8     │ 0.9     │ 0.8     │ 0.5     │ 0.1     │ 0.7     │ 0.6     │ 0.3     │ 0.8     │ 0.2     │ 0.8     │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ grade_4 │ 0.5     │ 0.9     │ 0.1     │ 0.9     │ 0.6     │ 0.8     │ 0.6     │ 0.5     │ 0.2     │ 0.8     │ 0.1     │ 0.8     │
+╘═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╛
+        """
+        old_scope = self.scope()[::-1].copy()
```

Comment:
@kislayabhi Add checks to raise an exception if `new_order` has some variable that is not in the actual factor.